hydra-node/src/Hydra/Chain/Direct/GenesisParameters.hs

```diff
@@ -0,0 +1,75 @@
+{-# LANGUAGE RecordWildCards #-}
+
+module Hydra.Chain.Direct.GenesisParameters where
+
+import Hydra.Prelude
+
+import Cardano.Ledger.BaseTypes (boundRational)
+import qualified Cardano.Ledger.BaseTypes as Ledger
+import Cardano.Ledger.Shelley.API (emptyGenesisStaking)
+import Cardano.Ledger.Shelley.PParams
+import Hydra.Cardano.Api
+import Hydra.Cardano.Api.Prelude (toShelleyLovelace, toShelleyNetwork)
+
+newtype GenesisTranslationError = GenesisTranslationError String deriving (Eq, Show)
+
+instance Exception GenesisTranslationError
+
+toShelleyGenesis :: GenesisParameters -> IO (ShelleyGenesis LedgerEra)
+toShelleyGenesis genesis =
+  case toShelleyGenesisMaybe genesis of
+    Nothing -> throwIO $ GenesisTranslationError "Could not translate GenesisParameters to ShelleyGenesis ShelleyEra"
+    (Just shelleyGenesis) -> pure shelleyGenesis
```

Comment:
No need to store a `String` in this error. It's already clear from the name. If you think you want to give more info.. make the name better.

Also, you should use `MonadThrow m` instead of ``IO` here as this gracefully falls back when used from `Maybe` for example.

Also, when throwing in our code: Add documentation! See also these best practices https://github.com/input-output-hk/hydra/wiki/Coding-Standards#exceptions